buildSrc/src/main/kotlin/Project.kt

```diff
@@ -2,7 +2,7 @@ object AndroidVersions {
   const val minSdkVersion = 21
   const val minAndroidAutoSdkVersion = 23
   const val targetSdkVersion = 30
-  const val compileSdkVersion = 30
+  const val compileSdkVersion = 31
```

Comment:
I think we should double check the bump of compile sdk version, could be part of the gradle version bump PR? cc @ank27  https://github.com/mapbox/mapbox-maps-android/pull/1118

Reply:
not sure if upstream SDKs have already bumped the compile SDK version, would be good to double check and verify before merging to main, since it will affect all the other modules as well.

Reply:
I have been rebasing, linking the other thread we had a month ago https://github.com/mapbox/mapbox-maps-android/pull/1004#discussion_r772684314

Is there a concern here? seems like we're good to pump it.

Reply:
This change probably will require downstream users to bump their compile SDK version as well, it's a big enough bump, where I think we should do it together in https://github.com/mapbox/mapbox-maps-android/pull/1118 with appropriate changelog/ PR description.

Reply:
thinking it over again, we could just bump the compileSdkVersion for Android Auto extension, e.g. `const val compileAndroidAutoSdkVersion = 31`.

As this is an Android Auto specific requirement and Android Auto extension is shipped as a separate binary that's not bundled with the Maps SDK by default. wdyt?

Reply:
Ok adding android auto version for this 👍

Reply:
also running into failures when separating
https://app.circleci.com/pipelines/github/mapbox/mapbox-maps-android/4230/workflows/0c0bc7b6-48c7-42ab-85d7-48f4737efba2/jobs/33055

Reply:
Was an issue targeting 31. Might have been a roboelectric issue